iib/workers/tasks/build_merge_index_image.py

```diff
@@ -0,0 +1,278 @@
+    """
+    Get a list of to-be-deprecated bundles based on the data from the deprecation list.
+
+    :param list bundles: list of bundles to apply the filter on.
+    :param list deprecation_list: list of deprecated bundle pull specifications (digest).
+    :return: bundles which are to be deprecated.
+    :rtype: list
+    """
+    deprecate_bundles = []
+    for bundle in bundles:
+        if '@sha256:' in bundle['bundlePath']:
+            resolved_bundle = bundle['bundlePath']
+        else:
+            resolved_bundle = _get_resolved_image(bundle['bundlePath'])
+        if resolved_bundle in deprecation_list:
+            deprecate_bundles.append(resolved_bundle)
+
+    log.info(
+        'Bundles that will be deprecated from the index image: %s', ', '.join(deprecate_bundles)
+    )
+    return deprecate_bundles
+
+
+def _deprecate_bundles(
+    bundles, base_dir, binary_image, from_index, overwrite_target_index_token=None,
+):
+    """
+    Deprecate specified bundles from index image. Only Dockerfile is created, no build is performed.
+
+    :param list bundles: pull specifications of bundles to deprecate.
+    :param str base_dir: base directory where operation files will be located.
+    :param str binary_image: binary image to be used by the new index image.
+    :param str from_index: index image, from which the bundles will be deprecated.
+    :param str overwrite_target_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required for non-privileged users to use
+        ``overwrite_target_index``. The format of the token must be in the format "user:password".
+    """
+    cmd = [
+        'opm',
+        'index',
+        'deprecatetruncate',
+        '--generate',
+        '--binary-image',
+        binary_image,
+        '--from-index',
+        from_index,
+        '--bundles',
+        ','.join(bundles),
+    ]
+    # TODO catch error of default channel deletion
+    with set_registry_token(overwrite_target_index_token, from_index):
+        run_cmd(
+            cmd, {'cwd': base_dir}, exc_msg='Failed to deprecate the bundles',
+        )
+
+
+@app.task
+@request_logger
+def handle_merge_request(
+    binary_image,
+    source_from_index,
+    deprecation_list,
+    request_id,
+    target_index=None,
+    overwrite_target_index=False,
+    overwrite_target_index_token=None,
+):
+    """
+    Coordinate the work needed to merge old (N) index image with new (N+1) index image.
+
+    :param str binary_image: the pull specification of the container image where the opm binary
+        gets copied from.
+    :param str source_from_index: pull specification to be used as base for building the new
+        index image.
+    :param str target_index: pull specification of content stage index image for the
+        corresponding target index image.
+    :param list deprecation_list: List of deprecated bundles for the target index image tag.
```

Comment:
To be consistent with the rest of the docstring.
```suggestion
    :param list deprecation_list: list of deprecated bundles for the target index image tag.
```